internal/libyaml/reader.go

```diff
@@ -23,15 +23,25 @@
 package libyaml
 
 import (
+	"errors"
+	"fmt"
 	"io"
 )
 
 // Set the reader error and return 0.
 func (parser *Parser) setReaderError(problem string, offset int, value int) bool {
-	parser.ErrorType = READER_ERROR
-	parser.Problem = problem
-	parser.ProblemOffset = offset
-	parser.ProblemValue = value
+	return parser.directlySetReaderError(
+		errors.New(problem),
+		offset, value,
+	)
```

Comment:
```suggestion
	parser.Error = v&ReaderError{ 
                 Offset: offset, 
                 Value:  value, 
                 Err:    errors.New(problem),
         }
	return false
```

See https://github.com/yaml/go-yaml/pull/177#discussion_r2500815112